contracts/utils/RateLimitedReplenishable.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {RateLimited} from "./RateLimited.sol";
+
+/// @notice Rate Limited Contract that allows replenishment of buffer
+/// @author Fei Protocol
+abstract contract RateLimitedReplenishable is RateLimited {
+    constructor(
+        uint256 _maxRateLimitPerSecond,
+        uint256 _rateLimitPerSecond,
+        uint256 _bufferCap,
+        bool _doPartialAction
+    )
+        RateLimited(
+            _maxRateLimitPerSecond,
+            _rateLimitPerSecond,
+            _bufferCap,
+            _doPartialAction
+        )
+    {}
+
+    /// @notice function to replenish buffer
+    /// @param amount to increase buffer by if under buffer cap
+    function _replenishBuffer(uint256 amount) internal {
```

Comment:
Can you override redeem and add a call to this inside of it as well?

Reply:
why would I override redeem to add this functionality in? I'll just add this logic straight into the redeem function itself.

Reply:
Right, that's actually what I meant, thank you!